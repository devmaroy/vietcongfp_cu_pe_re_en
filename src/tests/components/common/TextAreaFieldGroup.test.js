import React from 'react';
import { shallow } from 'enzyme';
import TextAreaFieldGroup from '../../../components/common/TextAreaFieldGroup';

test( 'should correctly render TextAreaFieldGroup only with required props', () => {
    const wrapper = shallow(
        <TextAreaFieldGroup 
            name="body"
            value="Some text here"
            onChange={ () => {} }
        />
    );

    expect( wrapper ).toMatchSnapshot();
});

test( 'should correctly render TextAreaFieldGroup with all props', () => {
    const wrapper = shallow(
        <TextAreaFieldGroup 
            name="body"
            value="Some"
            placeholder="Enter your message"
            onChange={ () => {} }
            info="This message will be under posts"
            error="Please enter at least 10 characters"
            disabled={ false }
        />
    );

    expect( wrapper ).toMatchSnapshot();
});