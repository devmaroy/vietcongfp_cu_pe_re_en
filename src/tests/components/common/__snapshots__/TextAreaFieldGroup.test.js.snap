// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render TextAreaFieldGroup only with required props 1`] = `
<div>
  <textarea
    name="body"
    onChange={[Function]}
    value="Some text here"
  />
</div>
`;

exports[`should correctly render TextAreaFieldGroup with all props 1`] = `
<div>
  <textarea
    disabled={false}
    name="body"
    onChange={[Function]}
    placeholder="Enter your message"
    value="Some"
  />
  <small>
    This message will be under posts
  </small>
  <div>
    Please enter at least 10 characters
  </div>
</div>
`;
